gcloud/pubsub/_gax.py

```diff
@@ -0,0 +1,173 @@
+
+            :type project: string
+            :param project: project ID
+
+            :rtype: tuple, (list, str)
+            :returns: list of ``Topic`` resource dicts, plus a
+                    "next page token" string:  if not None, indicates that
+                    more topics can be retrieved with another call (pass that
+                    value as ``page_token``).
+            """
+            options = CallOptions(is_page_streaming=False)
+            path = 'projects/%s' % (project,)
+            response = self._gax_api.list_topics(path, options)
+            topics = [{'name': topic_pb.name} for topic_pb in response.topics]
+            return topics, response.next_page_token
+
+        def topic_create(self, topic_path):
+            """API call:  create a topic
+
+            See:
+            https://cloud.google.com/pubsub/reference/rest/v1/projects.topics/create
+
+            :type topic_path: string
+            :param topic_path: fully-qualfied path of the new topic, in format
+                               ``projects/<PROJECT>/topics/<TOPIC_NAME>``.
+
+            :rtype: dict
+            :returns: ``Topic`` resource returned from the API.
+            :raises: :exc:`gcloud.exceptions.Conflict` if the topic already 
+                     exists
+            """
+            try:
+                topic_pb = self._gax_api.create_topic(topic_path)
+            except GaxError:
+                raise Conflict(topic_path)
+            return {'name': topic_pb.name}
+
+        def topic_get(self, topic_path):
+            """API call:  retrieve a topic
+
+            See:
+            https://cloud.google.com/pubsub/reference/rest/v1/projects.topics/get
+
+            :type topic_path: string
+            :param topic_path: fully-qualfied path of the topic, in format
+                            ``projects/<PROJECT>/topics/<TOPIC_NAME>``.
+
+            :rtype: dict
+            :returns: ``Topic`` resource returned from the API.
+            :raises: :exc:`gcloud.exceptions.NotFound` if the topic does not
+                     exist
+            """
+            try:
+                topic_pb = self._gax_api.get_topic(topic_path)
+            except GaxError:
+                raise NotFound(topic_path)
+            return {'name': topic_pb.name}
+
+        def topic_delete(self, topic_path):
+            """API call:  delete a topic
+
+            See:
+            https://cloud.google.com/pubsub/reference/rest/v1/projects.topics/create
+
+            :type topic_path: string
+            :param topic_path: fully-qualfied path of the new topic, in format
+                               ``projects/<PROJECT>/topics/<TOPIC_NAME>``.
+
+            :rtype: dict
+            :returns: ``Topic`` resource returned from the API.
+            """
+            try:
+                self._gax_api.delete_topic(topic_path)
+            except GaxError:
+                pass
```

Comment:
The only place we ignore 404s is in `storage.Client.lookup_bucket`, which has a sibling `get_bucket` that actually raises the exception.